src/DurableTask.SqlServer.AzureFunctions/SqlMetricsProvider.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace DurableTask.SqlServer.AzureFunctions
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    class SqlMetricsProvider
+    {
+        readonly SqlOrchestrationService service;
+        readonly int? previousWorkerCount;
+
+        public SqlMetricsProvider(SqlOrchestrationService service, int? previousWorkerCount = null)
+        {
+            this.service = service;
+            this.previousWorkerCount = previousWorkerCount;
+        }
+
+        public async Task<SqlScaleMetric> GetMetricsAsync()
+        {
+            // GetRecommendedReplicaCountAsync will write a trace if the recommendation results
+            // in a worker count that is different from the worker count we pass in as an argument.
+            int recommendedReplicaCount = await this.service.GetRecommendedReplicaCountAsync(
+                this.previousWorkerCount,
+                CancellationToken.None);
+
+            return new SqlScaleMetric { RecommendedReplicaCount = recommendedReplicaCount };
```

Comment:
I moved this code from `GetMetricsAsync()` in `SqlScaleMonitor()`.

Once I moved it here, `previousWorkerCount` is only set at initialization.

I looked at where `previousWorkerCount` is assigned [here ](https://github.com/microsoft/durabletask-mssql/blob/f4527f602bc53376496618f616977474903fc122/src/DurableTask.SqlServer.AzureFunctions/SqlScaleMonitor.cs#L62) and looks like the value comes from `ScaleStatusContext.WorkerCount` [here ](https://github.com/microsoft/durabletask-mssql/blob/f4527f602bc53376496618f616977474903fc122/src/DurableTask.SqlServer.AzureFunctions/SqlScaleMonitor.cs#L48) or [here](https://github.com/microsoft/durabletask-mssql/blob/f4527f602bc53376496618f616977474903fc122/src/DurableTask.SqlServer.AzureFunctions/SqlScaleMonitor.cs#L52). @cgillum, is there any other way to get the current worker count? I'm not sure I understand how the `GetScaleStatus` methods are invoked. Thanks!

Reply:
I'm not 100% sure, but I assume `GetScaleStatus` methods are invoked by the Functions host. Presumably the Functions host already knows about how many workers there are.